components/sidebar_right/index.js

```diff
@@ -22,12 +25,12 @@ import SidebarRight from './sidebar_right.jsx';
 
 function mapStateToProps(state) {
     const rhsState = getRhsState(state);
-
+    const channel = getCurrentChannel(state) || {};
```

Comment:
We can't be possibly returning object literals like `{}` from this function since that can cause `SidebarRight` to re-render constantly if the current channel is empty. You're already null-checking `channel` in `SidebarRight`, so this can just be replaced with
```suggestion
    const channel = getCurrentChannel(state);
```
safely.